service/matching/matcher.go

```diff
@@ -37,7 +38,8 @@ import (
 // that drains backlog from db. Consumers are the task list pollers
 type TaskMatcher struct {
 	// synchronous task channel to match producer/consumer
-	taskC chan *InternalTask
+	taskC         chan *InternalTask
+	isolatedTaskC map[string]chan *InternalTask
```

Comment:
Do you mind adding a small comment specifying the key to the string (zone / isolation-group I assume), but worth probably just makign that clear here